### PR TITLE
refactor(docs) improved & refactored docs

### PR DESCRIPTION
**Pull request by hoebbelsB**
improved and refactored docs.
- created docs folder
- moved API doc to own file
- moved USAGE doc to own file
- improved description and general appearance

### CODE REVIEW

File: libs/state/README.md [hunk 1]
```diff
@@ -2,28 +2,25 @@
...
+RxState is a light-weight, flexible, strongly typed and tested tool dedicated to reduce the complexity of managing component state in angular.
 
 ![state logo](https://raw.githubusercontent.com/BioPhoton/rx-angular/master/libs/state/images/state_logo.png)
 
 ## Description
 
-In most of the component-oriented applications, there is the need to structure container components.
-Even after a well thought refactoring into more display components and grouping logic into responsibilities it's always hard to handle.
+Developing **reactive**, **complex** and **robust** user interfaces confronts developers with manging an overwhelmingly huge amount
+events of different types and sources _aka_ **State Management**.
```
Comment (BioPhoton) [inline libs/state/README.md:14]: The sentence is confuded IMHO

File: libs/state/README.md [hunk 2]
```diff
@@ -2,28 +2,25 @@
...
 
-The data structure you manage inside these components is only here for their very component. Not for any other components.
-This data structure appears with the component and disappears when the component is removed.
+There are plenty solutions available for managing state on a **global level** (Akita, NgRx, NgXs, ...). All of
```
Comment (BioPhoton) [inline libs/state/README.md:16]:
```suggestion
There are plenty of solutions available for managing state on a **global level** (Akita, NgRx, NgXs, ...). All of
```

File: libs/state/README.md [hunk 3]
```diff
@@ -2,28 +2,25 @@
...
+them **not targeting** managing state on **component level**.
 
-This is a good example of component state.
+Thats where `RxState` comes into place.
```
Comment (BioPhoton) [inline libs/state/README.md:19]:
```suggestion
That's where `@rx-angular/state` comes into place.
```

File: libs/state/README.md [hunk 4]
```diff
@@ -2,28 +2,25 @@
...
+The **intuitive API** specifically designed for managing **component state**, **automatic subscription handling** and
+**lifecycle integration** making `RxState` the **perfect fit** for handling **component state** in Angular.
```
Comment (BioPhoton) [inline libs/state/README.md:21]:
```suggestion
**lifecycle integration** making `@rx-angular/state` the **perfect fit** for handling **component state** in Angular.
```

File: libs/state/README.md [hunk 5]
```diff
@@ -35,231 +32,19 @@ It is an easy, flexible, fully typed and tested reactive component state managem
...
-import { fromEvent } from 'rxjs/observable';
-state.hold(btnClick$, clickEvent => console.log(clickEvent));
-```
-
----
-
-## Usage in Angular applications
-
-### Basic Setup
+The recommended way of using the RxState is by **providing** a local instance bound to the component lifecycle.
```
Comment (BioPhoton) [inline libs/state/README.md:47]:
```suggestion
The recommended way of using the `RxState` service is by **providing** a local instance bound to the component lifecycle.
```

File: libs/state/README.md [hunk 6]
```diff
@@ -278,16 +63,16 @@ export class StatefulComponent {
 }
 ```
 
-In very small components you can also extend to keep the api short.
+### Extend (minimal)
 
-**Extend from the `RxState` inside a `Component`, `Directive` or `Service`**
+If you wish, there is also the possibility of **extending** the `RxState` service. This is generally
```
Comment (BioPhoton) [inline libs/state/README.md:68]: I think both ways are recommended but for different use cases. We could rephrase it to be "equally recommended" for both situations.

File: libs/state/README.md (hunk 1 above)
Comment (hoebbelsB) [reply libs/state/README.md:14]: will split up the sentence in mangaging/transforming state and handling/bundling events

File: libs/state/README.md (hunk 5 above)
Comment (hoebbelsB) [inline libs/state/README.md]: get rid of 'recommended'. use e.g. 'one way of using...'

File: libs/state/README.md (hunk 6 above)
Comment (hoebbelsB) [reply libs/state/README.md:68]: yes, it's just another way with different pros and cons.

File: libs/state/README.md [hunk 7]
```diff
@@ -35,231 +32,19 @@ It is an easy, flexible, fully typed and tested reactive component state managem
...
-
-**Access by providing rxjs operators**
-
-```typescript
-const state = new RxState<{ loo: { bar: string } }>();
-state.setState({ bar: 'boo' });
-
-const customProp$ = state.select(map(state => state?.loo?.bar));
-customProp$.subscribe(console.log); // 'boo'
+##API
```
Comment (hoebbelsB) [inline libs/state/README.md:39]:
```suggestion
## API
```

File: libs/state/README.md [hunk 8]
```diff
@@ -278,16 +63,16 @@ export class StatefulComponent {
 }
 ```
 
-In very small components you can also extend to keep the api short.
+### Extend (minimal)
```
Comment (hoebbelsB) [inline libs/state/README.md:66]:
```suggestion
### Inherit
```

File: libs/state/README.md [hunk 9]
```diff
@@ -35,231 +32,19 @@ It is an easy, flexible, fully typed and tested reactive component state managem
...
-```typescript
-const sideEffect$ = btnClick$.pipe(
-  tap(clickEvent => this.store.dispatch(loadAction()))
-);
-const subscription = sideEffect$.subscribe();
-subscription.unsubscribe();
-```
-
-If you would hold to achieve the same thing it would look like this:
+### Provide (recommended)
```
Comment (hoebbelsB) [inline libs/state/README.md:45]:
```suggestion
### Compose
```

File: libs/state/README.md [hunk 10]
```diff
@@ -35,231 +32,19 @@ It is an easy, flexible, fully typed and tested reactive component state managem
...
-import { fromEvent } from 'rxjs/observable';
-state.hold(btnClick$, clickEvent => console.log(clickEvent));
-```
-
----
-
-## Usage in Angular applications
-
-### Basic Setup
+The recommended way of using the `RxState` service is by **providing** a local instance bound to the component lifecycle.
```
Comment (hoebbelsB) [inline libs/state/README.md:47]:
```suggestion
One way of using the `RxState` service is by `providing` a local instance bound to the component lifecycle.
```

File: libs/state/README.md [hunk 11]
```diff
@@ -278,16 +63,16 @@ export class StatefulComponent {
...
 ```
 
-In very small components you can also extend to keep the api short.
+### Extend (minimal)
 
-**Extend from the `RxState` inside a `Component`, `Directive` or `Service`**
+If you wish, there is also the possibility of **extending** the `RxState` service. This is generally
+not the recommended way but indeed useful for very small components.
+
+**Extend from the `RxState` inside a `Component`, `Directive`**
```
Comment (hoebbelsB) [inline libs/state/README.md:71]:
```suggestion
Extend from the `RxState` inside a `Component`, `Directive`
```

File: libs/state/README.md [hunk 12]
```diff
@@ -298,143 +83,49 @@ export class StatefulComponent extends RxState<{ foo: number }> {
...
-### Connecting services
+## Extended Setup
 
-**Connect state slices from third party services (e.g. NgRx `Store`) or trigger them from side-effects**
+If you strive for a more sophisticated **separation of concerns** you can `extend` the `RxState` in a
+locally provided `Service`.
 
-Many people have problems combining observables with the component state in a clean way.
-er a usecase where the @ngrx/store gets connected to the local state:
+**Create a local `Service` by `extending` the `RxState`**
```
Comment (hoebbelsB) [inline libs/state/README.md:91]:
```suggestion
Create a local `Service` by `extending` the `RxState`
```

File: libs/state/README.md [hunk 13]
```diff
@@ -298,143 +83,49 @@ export class StatefulComponent extends RxState<{ foo: number }> {
...
+    super();
   }
-  constructor(private state: RxState<{ count: number }>) {}
 }
 ```
 
-### Output Property Bindings
-
-**Combining `Output` bindings directly from RxState**
+**`Provide` the `Service` inside the using `Component` or `Directive`**
```
Comment (hoebbelsB) [inline libs/state/README.md:107]:
```suggestion
`Provide` the `Service` inside the using `Component` or `Directive`
```

File: libs/state/README.md [hunk 14]
```diff
@@ -35,231 +32,19 @@ It is an easy, flexible, fully typed and tested reactive component state managem
...
-```
-
----
-
-## Usage in Angular applications
-
-### Basic Setup
+The recommended way of using the `RxState` service is by **providing** a local instance bound to the component lifecycle.
 
 **Provide the `RxState` inside a `Component` or `Directive` and `Inject` it**
```
Comment (hoebbelsB) [inline libs/state/README.md:49]:
```suggestion
Provide the `RxState` inside a `Component` or `Directive` and `Inject` it
```
